### PR TITLE
Convert NuGet.PackageManagement.VisualStudio to an SDK based project

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2038

Regression? Last working version:

## Description

Make our solution work smoother in VS by using SDK based projects.

I had to move 1 type because we're depending on a reference not available in an SDK project.
I'd even argue that the type belongs there anyways. 

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A  Tests already exist.

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/NuGet.PackageManagement.VisualStudio.csproj [hunk 1]
```diff
@@ -1,297 +1,61 @@
...
     <Reference Include="System" />
     <Reference Include="System.ComponentModel.Composition" />
     <Reference Include="System.Core" />
     <Reference Include="System.IO.Compression" />
+    <Reference Include="System.Runtime.Caching" />
     <Reference Include="System.Runtime.Serialization" />
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="System.Xml.Linq" />
     <Reference Include="WindowsBase" />
-    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL" />
```
Comment (dominoFire) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/NuGet.PackageManagement.VisualStudio.csproj:30]: How did you migrate both projects? Did you use try-convert or upgrade-assistant?

Is this Reference imported automatically in an SDK-style project?

**Review by dominoFire — COMMENTED** [review]
One question. Thank you!

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/NuGet.PackageManagement.VisualStudio.csproj (hunk 1 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/NuGet.PackageManagement.VisualStudio.csproj:30]: If I compile test project with `dotnet build` this reference is not found.

image

Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/NuGet.PackageManagement.VisualStudio.csproj:30]: Oh, I see, this is why you moved `IPSNuGetProjectContext` to NuGet.Console, so this project can get rid of System.Management.Automation reference.

**Review by dominoFire — APPROVED** [review]

**Review by erdembayar — APPROVED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/NuGet.PackageManagement.VisualStudio.csproj (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/NuGet.PackageManagement.VisualStudio.csproj:30]: Yep, that's exactly it. 
I migrated everything manually.

**Pull request merged**